### DISCUSSION TITLE
Logging model event using queue

### DISCUSSION THREAD

**Discussion by laravelbap** [spatie/laravel-activitylog#842] [Q&A] [answered]
Hi,

When i have a lot of activities in DB (>50k) query that inserts data into "insert into `activity_log` (`log_name`,"....) takes about **181.83ms** when I remove most of entries in activity log table it takes **4ms**. 

Is it possible to set up a logging model to use queue or you Guys/Girs have some advice for me?

**Comment by Gummibeer** [comment, marked as answer]
I would recommend you to check if/how you can optimize the indexes and if your database (server) has potential hard drive, IOPS, CPU or similar limits.
We've run projects with millions of records in activities table without any decrease in insert performance.
If you use MySQL there are several articles regarding InnoDB and indexes if you search for "MySQL slow insert query".

**Reply by laravelbap** [reply]
Thanks, i will check.
